tools/pythonpkg/tests/fast/test_replacement_scan.py

```diff
@@ -152,10 +292,11 @@ def test_replacement_disabled(self):
 
         con.execute("set enable_external_access=false")
 
-        with pytest.raises(
-            duckdb.InvalidInputException, match='Attempting to execute an unsuccessful or closed pending query result'
-        ):
-            res = rel.fetchall()
+        # Since we cache the replacement scans as CTEs, disabling the external access inbetween creation
+        # and execution has no effect, we might want to change that by keeping track of which CTEs we have added
+        # and removing them if `enable_external_access` is set
```

Comment:
This was added recently and is something we now have to think about it
Do we think it's worth it to do this or is this corner case so esoteric that we don't care?

Reply:
I think this is fine, `enable_external_access` only blocks replacement scans after it is set